api/tests/unit/domain/usecases/authenticate-external-user_test.js

```diff
@@ -0,0 +1,438 @@
+      });
+
+      it('should throw UserShouldChangePasswordError', async () => {
+        // given
+        const oneTimePassword = 'Azerty123*';
+        const { user } = createUserWithValidCredentialsWhoShouldChangePassword({
+          authenticationService,
+          userRepository,
+        });
+
+        const externalUserToken = 'EXTERNAL_USER_TOKEN';
+        const externalIdentifier = 'EXTENAL_IDENTIFIER';
+        createValidConditionsForAddingGarAuthenticationMethod({
+          user,
+          externalUserToken,
+          samlId: externalIdentifier,
+          tokenService,
+          userRepository,
+          authenticationMethodRepository,
+        });
+
+        // when
+        const error = await catchErr(authenticateExternalUser)({
+          username: user.email,
+          password: oneTimePassword,
+          externalUserToken,
+          expectedUserId: user.id,
+          tokenService,
+          authenticationService,
+          authenticationMethodRepository,
+          userRepository,
+        });
+
+        // then
+        expect(error).to.be.an.instanceOf(UserShouldChangePasswordError);
+      });
+    });
+
+  });
+
+  context('when credentials are invalid', () => {
+
+    it('should reject when user not found', async () => {
+      // given
+      const unknownUserEmail = 'foo@example.net';
+      const password = 'Azerty123*';
+
+      authenticationService.getUserByUsernameAndPassword.withArgs({
+        username: unknownUserEmail,
+        password,
+        userRepository,
+      }).rejects(new UserNotFoundError());
+
+      // when
+      const error = await catchErr(authenticateExternalUser)({
+        username: unknownUserEmail,
+        password,
+        tokenService,
+        authenticationService,
+        userRepository,
+      });
+
+      // then
+      expect(error).to.be.an.instanceOf(MissingOrInvalidCredentialsError);
+    });
+
+    it('should reject when password does not match', async () => {
+      // given
+      const email = 'foo@example.net';
+      const invalidPassword = 'oups123*';
+
+      authenticationService.getUserByUsernameAndPassword.withArgs({
+        username: email,
+        password: invalidPassword,
+        userRepository,
+      }).rejects(new PasswordNotMatching());
+
+      // when
+      const error = await catchErr(authenticateExternalUser)({
+        username: email,
+        password: invalidPassword,
+        tokenService,
+        authenticationService,
+        userRepository,
+      });
+
+      // then
+      expect(error).to.be.an.instanceOf(MissingOrInvalidCredentialsError);
+    });
+  });
+
+});
+
+function createUserWithValidCredentials({
+  password = 'Azerty123*',
+  authenticationService,
+  userRepository,
+}) {
+  const email = 'john.doe@example.net';
+  const emailAuthenticationMethod = domainBuilder.buildAuthenticationMethod({
```

Comment:
```suggestion
  const pixAuthenticationMethod = domainBuilder.buildAuthenticationMethod({
```

Par ailleurs ça build une authenticationMethod GAR actuellement

Reply:
Corrigé et objet du domaine introduit :tada: